src/main/java/com/powsybl/openloadflow/sensi/AbstractDcSensitivityAnalysis.java

```diff
@@ -0,0 +1,368 @@
+import com.powsybl.openloadflow.network.util.LoadActivePowerDistributionStep;
+import com.powsybl.openloadflow.network.util.ParticipatingElement;
+import com.powsybl.sensitivity.SensitivityFactor;
+import com.powsybl.sensitivity.SensitivityValue;
+import com.powsybl.sensitivity.factors.BranchFlowPerInjectionIncrease;
+import com.powsybl.sensitivity.factors.BranchFlowPerPSTAngle;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Gaël Macherel <gael.macherel@artelys.com>
+ */
+public abstract class AbstractDcSensitivityAnalysis extends AbstractSensitivityAnalysis {
+
+    protected AbstractDcSensitivityAnalysis(final MatrixFactory matrixFactory) {
+        super(matrixFactory);
+    }
+
+    protected static class LazyConnectivity {
+        private GraphDecrementalConnectivity<LfBus> connectivity = null;
+        private LfNetwork network;
+
+        public LazyConnectivity(LfNetwork lfNetwork) {
+            network = lfNetwork;
+        }
+
+        public GraphDecrementalConnectivity<LfBus> getConnectivity() {
+            if (connectivity != null) {
+                return connectivity;
+            }
+            connectivity = new NaiveGraphDecrementalConnectivity<>(LfBus::getNum); // FIXME: use EvenShiloach
+            for (LfBus bus : network.getBuses()) {
+                connectivity.addVertex(bus);
+            }
+            for (LfBranch branch : network.getBranches()) {
+                connectivity.addEdge(branch.getBus1(), branch.getBus2());
+            }
+            return connectivity;
+        }
+    }
+
+    static class SensitivityFactorWrapped {
+        // Wrap factors in another class to have instant access to their branch, and their equation term
+        private final SensitivityFactor factor;
+
+        private final LfBranch functionBranch;
+
+        private final String functionBranchId;
+
+        private final ClosedBranchSide1DcFlowEquationTerm equationTerm;
+
+        private Double predefinedResult = null;
+
+        private Double functionReference;
+
+        public SensitivityFactorWrapped(SensitivityFactor factor, LfNetwork lfNetwork, EquationSystem equationSystem) {
+            this.factor = factor;
+            if (factor instanceof BranchFlowPerInjectionIncrease) {
+                functionBranch = lfNetwork.getBranchById(((BranchFlowPerInjectionIncrease) factor).getFunction().getBranchId());
+            } else if (factor instanceof  BranchFlowPerPSTAngle) {
+                functionBranch = lfNetwork.getBranchById(((BranchFlowPerPSTAngle) factor).getFunction().getBranchId());
+            } else {
+                throw new UnsupportedOperationException("Only factors of type BranchFlowPerInjectionIncrease and BranchFlowPerPSTAngle are supported");
```

Comment:
You already check that earlier in your code. Do you really need that line that will not be covered?